executor/partition_table_test.go

```diff
@@ -198,3 +198,13 @@ partition p2 values less than (10))`)
 	tk.MustExec("insert into p values (1,3), (3,4), (5,6), (7,9)")
 	tk.MustQuery("select * from p use index (idx)").Check(testkit.Rows("1 3", "3 4", "5 6", "7 9"))
 }
+
+func (s *globalIndexSuite) TestIssue21732(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
```

Comment:
In the unit test, we use @@tidb_partition_prune_mode = 'dynaimc-only' as default.
It should also check the case when tidb_partition_prune_mode is 'static-only'...